starky/src/proof.rs

```diff
@@ -46,15 +52,57 @@ impl<F: RichField + Extendable<D>, C: GenericConfig<D, F = F>, const D: usize> S
     }
 }
 
+/// Circuit version of [`StarkProof`].
+/// Merkle caps and openings that form the proof of a single STARK.
+#[derive(Clone, Debug, PartialEq, Eq)]
 pub struct StarkProofTarget<const D: usize> {
+    /// `Target` for the Merkle cap if LDEs of trace values.
     pub trace_cap: MerkleCapTarget,
+    /// `Target` for the Merkle cap of LDEs of lookup helper and CTL columns.
     pub auxiliary_polys_cap: Option<MerkleCapTarget>,
+    /// `Target` for the Merkle cap of LDEs of quotient polynomial evaluations.
     pub quotient_polys_cap: MerkleCapTarget,
+    /// `Target`s for the purported values of each polynomial at the challenge point.
     pub openings: StarkOpeningSetTarget<D>,
+    /// `Target`s for the batch FRI argument for all openings.
     pub opening_proof: FriProofTarget<D>,
 }
 
 impl<const D: usize> StarkProofTarget<D> {
+    /// Serializes a STARK proof.
+    pub fn to_buffer(&self, buffer: &mut Vec<u8>) -> IoResult<()> {
+        buffer.write_target_merkle_cap(&self.trace_cap)?;
+        if let Some(poly) = &self.auxiliary_polys_cap {
+            buffer.write_bool(true)?;
+            buffer.write_target_merkle_cap(poly)?;
+        }
```

Comment:
Should there be an `else { buffer.write_bool(false)?; }` if my next comment below is true?

Reply:
Ah yeah good catch. Moved the boolean write outside of the `if` as we need it in both cases